components/appmgt-extensions/org.wso2.carbon.appmgt.mdm.wso2mdm.direct/src/main/java/org/wso2/carbon/appmgt/mdm/wso2mdm/direct/MDMOperationsImpl.java

```diff
@@ -0,0 +1,308 @@
+                }
+            }
+
+        } catch (AppManagementException | ApplicationManagementException e) {
+            log.error("Could not retrieve the 'DeviceManagementProviderService' by OSGI lookup. User [" + currentUser
+                    + "], tenant [" + tenantId + "] config params [" + configParams + "]", e);
+        }
+
+    }
+
+    private List<DeviceIdentifier> convertDeviceIdentifiers(String type, String[] params) {
+        List<DeviceIdentifier> result = new ArrayList<>();
+        if ("user".equals(type) || "role".equals(type)) {
+            for (String param : params) {
+                DeviceIdentifier deviceIdentifier = new DeviceIdentifier(param, type);
+                result.add(deviceIdentifier);
+            }
+        } else {
+            for (String param : params) {
+                String[] paramDevices = param.split(APPM_DEVICE_ID_CONCAT_SEPARATOR);
+                DeviceIdentifier deviceIdentifier = new DeviceIdentifier(paramDevices[0], paramDevices[1]);
+                result.add(deviceIdentifier);
+            }
+        }
+
+        return result;
+    }
+
+    private MobileAppBean createMobileApp(User currentUser, String action, App app, int tenantId, String type,
+            String[] params, HashMap<String, String> configParams) {
+
+        MobileAppBean mobileApp = new MobileAppBean();
+        Class mobileAppClazz = MobileAppBean.class;
+
+        Method[] methods = app.getClass().getMethods();
+
+        //Copies the AppM side mobile app properties to MDM mobile App. Uses reflection and annotation to perform data transfer
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(Property.class)) {
+                try {
+                    Object value = method.invoke(app);
+                    if (value != null) {
+                        String propName = method.getAnnotation(Property.class).name();
+                        propName = propName.substring(0, 1).toUpperCase() + propName.substring(1, propName.length());
+                        if (propName.equals("Type")) {
+                            value = MobileAppTypes.valueOf(String.valueOf(value).toUpperCase());
+                        }
+                        Method m2 = mobileAppClazz.getMethod("set" + propName, value.getClass());
+                        m2.invoke(mobileApp, value);
+                    }
+                } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+                    String errorMessage = "Illegal Action";
+                    if (log.isDebugEnabled()) {
+                        log.error(errorMessage, e);
+                    } else {
+                        log.error(errorMessage);
+                    }
+                }
+            }
+
+        }
+
+        Properties requestApp = new Properties();
+        requestApp.put(APPM_APP_TYPE, mobileApp.getType().name());
+
+        Properties properties = new Properties();
```

Comment:
We need to have all app metadata in the AppM side, taken from app binary, and process it here. then remove this hard-coded property setting. Create a JIRA in AppM to track this.